### DISCUSSION TITLE
Partial folding with TailwindCSS utilities

### DISCUSSION THREAD

**Discussion by pieroguerrero** [moalamri/vscode-inline-fold#49] [Q&A] [answered]
Hello everyone

First at all I want to congratulate you guys for this awesome and helpful plugin. 
Didn't know where to put this, maybe I just don't know how to use the plugin correctly.

I am a fan of TailwindCSS and I have a React component like this: 

![Screenshot from 2022-08-27 21-16-21](https://user-images.githubusercontent.com/26049605/187054618-0ac5b6ff-4df2-4007-8b8c-ef3865cabb6f.png)

However, the plugin only folds in this way:

![Screenshot from 2022-08-27 21-17-31](https://user-images.githubusercontent.com/26049605/187054644-628f0ba0-8136-4a13-8bc2-e175eef4a249.png)

Looks like the utility classes below are not being recognized properly: 
```
before:content-['']
after:content-['']
```
I tried removing all the blank spaces and line changes but the result is the same. The component is working correctly.
Any clue on this please?

Hope this helps to improve this plugin 👍🏼

**Comment by moalamri** [comment]
Hi,
I can't figure if you are using new lines between the className value or that is the editor breaking lines in the view?

**Reply by pieroguerrero** [reply]
Hi @moalamri 
They are the editor breaking lines, I haven't added any.

**Comment by moalamri** [comment, marked as answer]
@pieroguerrero 
It's because the regex is missing some of the content of the class value.
Try this regex: https://regex101.com/r/ECoxwG/1

Regex (escaped):
```jsonc
{
 "InlineFold.regex": "(class|className)=\"([a-zA-Z0-9-\\s\'\[\\]:]+)(?=\")"
}
```

Or through Settings UI:
`(class|className)="([a-zA-Z0-9-\s'\[\]:]+)(?=")`

Make sure regex group is set to the default value `2`.

**Reply by pieroguerrero** [reply]
Thank you so much @moalamri 
This resolved the miss-configuration. I also improved the regex since there was a className tag not being recognized properly:

`className="pb-[24px] sm:pb-0 text-[33px] sm:text-[58px] text-color-text-accent sm:font-light leading-[1.2] sm:text-opacity-90"`

The issue was with the "leading-[1.2]" utility class.

The new regex is:
`"([.a-zA-Z0-9-\s'\[\]:]+)(?=\")`

Cheers!
